packages/flutter_tools/lib/src/commands/build.dart

```diff
@@ -26,7 +28,12 @@ class BuildCommand extends FlutterCommand {
     addSubcommand(BuildAppBundleCommand(verboseHelp: verboseHelp));
     addSubcommand(BuildAotCommand(verboseHelp: verboseHelp));
     addSubcommand(BuildIOSCommand());
-    addSubcommand(BuildIOSFrameworkCommand());
+    addSubcommand(BuildIOSFrameworkCommand(
+      aotBuilder: AotBuilder(),
+      bundleBuilder: BundleBuilder(),
+      cache: globals.cache,
+      platform: globals.platform,
+    ));
```

Comment:
@jonahwilliams What do you think of this pattern?  I don't love that these objects are getting created even when some other build command is being run.

Reply:
Object creation is cheap, technical debt is expensive.

Reply:
Though if you feel particularly bad about any particular cases, you could create a FooFactory